gnd/src/main/java/com/google/android/gnd/ui/settings/SettingsFragment.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.settings;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout.LayoutParams;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.preference.Preference;
+import androidx.preference.Preference.OnPreferenceChangeListener;
+import androidx.preference.Preference.OnPreferenceClickListener;
+import androidx.preference.PreferenceFragmentCompat;
+import com.google.android.gnd.MainActivity;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.SettingsFragBinding;
+
+/**
+ * Fragment containing app preferences saved as shared preferences.
+ *
+ * <p>NOTE: It uses {@link PreferenceFragmentCompat} instead of {@link
+ * com.google.android.gnd.ui.common.AbstractFragment}, so dagger can't inject into it like it does
+ * in other fragments.
+ *
+ * <p>TODO: Create dagger module and support injection into this fragment.
+ */
+public class SettingsFragment extends PreferenceFragmentCompat
+    implements OnPreferenceChangeListener, OnPreferenceClickListener {
+
+  @Override
+  public View onCreateView(
+      @NonNull LayoutInflater inflater,
+      @Nullable ViewGroup container,
+      @Nullable Bundle savedInstanceState) {
+    super.onCreateView(inflater, container, savedInstanceState);
+    SettingsFragBinding binding = SettingsFragBinding.inflate(inflater, container, true);
+    binding.setLifecycleOwner(this);
+    ((MainActivity) getActivity()).setActionBar(binding.settingsToolbar, true);
+    return super.onCreateView(inflater, container, savedInstanceState);
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    // TODO: Set dynamically by calculating the actual height of toolbar
+    int topMargin = 108;
+
+    LayoutParams params = (LayoutParams) getListView().getLayoutParams();
+    params.setMargins(0, topMargin, 0, 0);
+    getListView().setLayoutParams(params);
```

Comment:
I tried to get the height of the view from binding. But it always returns 0 for the first time. Anyone knows why?

The reason we have to add top margin is to display the toolbar. There is no standard practice around how to add a toolbar because official docs recommend using the `PreferenceFragment` from within an Activity. In our case, the activity doesn't have a toolbar. Any guidelines or suggestions are most welcome!

Reply:
Removed